corehq/apps/users/models.py

```diff
@@ -230,6 +230,11 @@ class UserRole(Document):
     permissions = SchemaProperty(Permissions)
     is_archived = BooleanProperty(default=False)
 
+    @classmethod
+    @quickcache(['doc_id'], timeout=0)
+    def get(cls, doc_id):
```

Comment:
I'm not a huge fan of changing the default behavior of `SomeDocumentClass.get()` to possibly return stale results even if that staleness is isolated. I think I'd prefer to see this be `cached_get` or have a `strict` parameter that defaults to `True` as to preserve the expected behavior, and then override those instances that we want to cache with the caching version. Caching can introduce pretty hairy bugs, and I don't think it's a bad thing to force developers to think about when they actually want to use a cache versus not.

As one example, I think on the user list page you definitely want cached roles, but on the "edit user" page you might not.